service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -243,21 +276,26 @@ public List<String> validateMethod(PostMethodRequest methodRequest) {
     if (methodUrl == null || methodUrl.trim().isEmpty()) {
       errors.add("method_url is required");
     } else {
-      // verify that URL is valid, and it's host is supported
-      try {
-        URL url = new URI(methodUrl).toURL();
-        Pattern pattern =
-            Pattern.compile(
-                "^https?:\\/\\/(?:www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b(?:[-a-zA-Z0-9()@:%_\\+.~#?&\\/=]*)$");
-        boolean doesUrlMatchPattern = pattern.matcher(methodUrl).find();
-
-        if (!doesUrlMatchPattern) {
-          errors.add("method_url is invalid. URL doesn't match pattern format");
-        } else if (!SUPPORTED_URL_HOSTS.contains(url.getHost())) {
-          errors.add("method_url is invalid. Supported URI host(s): " + SUPPORTED_URL_HOSTS);
+      // we only verify if URL is valid for GitHub source here. For Dockstore methods a workflow
+      // path which is not completely a valid URL is sent as method url, and it's validity is
+      // checked while fetching the raw GitHub url for the workflow path
+      if (methodRequest.getMethodSource() == PostMethodRequest.MethodSourceEnum.GITHUB) {
+        // verify that URL is valid, and it's host is supported
```

Comment:
```suggestion
        // verify that URL is valid, and its host is supported
```

Reply:
Looks like this wasn't even your comment originally, so this is me being extra-double pedantic... 😆